Finished StatesRepositoryIT test class

gabichelsea opened the pull request:
Recreated pull request to persistence-layer-integration-test branch

Trello card link:
[https://trello.com/c/EzJ8BV0K](https://trello.com/c/EzJ8BV0K)

gabichelsea committed:
Finished StatesRepositoryIT test class

kbertalan commented:
As per there were some push issues at this task I have to summarize here the review findings:
- Move states sql into its own folder.
- Remove extra empty lines in both SQL and StatesRepositoryIT
- Do not commit changes which contains only white space changes, see 'Modify nothing' commit
- Use Java naming conventions for named constants
- checkValidationGivenProcedureAndGetApplicationListByCandidateNameAndSizeOfApplicationList method is used to setup test fixture/context in a Given phase, it should not contain any checks against those data which are known to be there as they are inserted by flyway or your states.sql
- checkValidationGivenProcedureAndGetApplicationListByCandidateNameAndSizeOfApplicationList returns a list, but your client code needs a single element, rewrite it to return a single element
- use isEmpty instead of size assertions for checking empty list/collections
- move getExpectedStateEntityXY calls to given phase and introduce local variable if needed
- move getExpectedSortingStateEntityListWithThreeElements call from checkSortingValidationWithThreeElements into test methods given phase

gabichelsea committed:
Fixed it

kbertalan commented on persistence/src/test/java/com/epam/rft/atsy/persistence/repositories/StatesRepositoryIT.java:
```diff
@@ -74,47 +69,38 @@ public void findTopByApplicationEntityOrderByStateIndexDescShouldNotFindStateFor
         StateEntity expectedStateEntity = getExpectedStateEntityForSingleState();
 
+        // When
+        StateEntity stateEntity = this.statesRepository.findByApplicationEntityOrderByStateIndexDesc(applicationEntity).get(ZEROTH_ELEMENT);
```
You don't assert that there are only a single state in the returned list.

kbertalan commented:
Issues remained:
- Use Java naming conventions for named constants
- Remove extra empty lines in both SQL and StatesRepositoryIT
- also commented in line for missing assertions

gabichelsea committed:
Fixed it:

Use Java naming conventions for named constants
Remove extra empty lines in both SQL and StatesRepositoryIT
You don't assert that there are only a single state in the returned list.
